### ISSUE TITLE
Tracker more accurantely tracks enemy moves and the level at which it had that move

### ISSUE THREAD

**Issue by UTDZac** [besteon/Ironmon-Tracker#202]
Version: 0.6.1

Currently only four moves are tracked for any enemy Pokemon. This can often times be inaccurate in a sense that encountering the same Pokemon at different levels show incomplete move information. Easiest when shown with an example, see blow.

![missing dream eater](https://user-images.githubusercontent.com/4258818/185767092-7ef7f679-89aa-404b-a5e5-f72b9a5ff71c.png)

Here you can see I've encountered the Forretress quite a few times, mostly around levels 3, 5, 6, and 8. It has four unique moves before level 8, then learns a fifth move at that level. This screen above shows the movelist for the level 8 Forretress, which is wrong because I'm looking at a level 5 Forretress. **It is missing the move Dream Eater (its first level 1 move)**.

My proposed solution is to track ALL unique moves of a Pokemon at all times. It would also associate / update the move with the lowest relevant level of the encounter Pokemon (internal tracked data information). This way the Tracker can more accurate display to the user the proper moves at the proper level, for any level of Pokemon that they run into.

### Example flow of this implementation:
1. User encounters the first Forretress at level 5. They scout all of the moves. The data stored in the tracker would then be:

| Move | Min Level Seen |
| --- | --- |
| Dream Eater | 5 |
| Steel Wing | 5 |
| Psychic | 5 |
| Fire Punch | 5 |

2. The user continues to check the grass and encounters the lv 8. The data stored in the tracker would then be:

| Move | Min Level Seen |
| --- | --- |
| Dream Eater | 5 |
| Steel Wing | 5 |
| Psychic | 5 |
| Fire Punch | 5 |
| Ember | 8 |

3. The user continues checking grass, encountering the lv3 Forretress. Data would be:

| Move | Min Level Seen |
| --- | --- |
| Dream Eater | 3 |
| Steel Wing | 3 |
| Psychic | 3 |
| Fire Punch | 3 |
| Ember | 8 |

### Additional Resulting Feature (new issue to be added later)
Once this is implemented, this allows us to provide the user with more information about any give Pokemon via the Pokemon Info Screen. When viewing the Pokemon info page and looking at "moves learned at a level" they can then click that area to see the full list of knowledge about the moves. This is basically the same info they would have if they were manually tracking the levels of moves for each encounter and updating them along the way (similar to the data tables above).

**Comment by UTDZac** [comment]
We can use this as an opportunity to capture additional move data and track it. For example, when you transform into a pokemon, then we can track each of their moves:
> `Tracker.TrackMove(moveId, opposingPokemon.level)` (x4)

We can do the same when we choose to actively use a Pokemon on our team, similar to how we do this for ability tracking

https://github.com/besteon/Ironmon-Tracker/blob/dc226f24dc7aaa35d3ff21706d180c83d4a42c80/ironmon_tracker/Battle.lua#L137-L138

**Comment by UTDZac** [comment]
Leaving a note here that this was significantly more challenging to solve than I had initially considered. I think since this is fairly low priority I will label it as such.

However, I will take this time to rework the way that moves are drawn onto the screen, instead turning them into borderless buttons. This cleans up a bit of logic in some places, and allows us to more easily create a "Enemy Info Screen" that a few people have requested. More on that at a later time.

**Comment by UTDZac** [comment]
I'm wanting to revisit this, but only after mGBA support is finished, as that has reworked a ton of how move data is built up to be displayed

Another user/viewer suggested this:
> Oh, one thing I noticed the other day Zac, any plans to do internal tracking for when moves are learned? i.e. you see an umbreon at level 40, then see one in the grass at level 20, while in combat with the level 20 mon you see the moves it used when you encountered it at level 40.

**Comment by UTDZac** [comment]
This is being worked on through this branch: https://github.com/besteon/Ironmon-Tracker/tree/utdzac/tracked-move-history

Ultimately going with a simpler solution that doesn't worry about what four moves are shown on the main Tracker Screen, but rather offering a way to see the full tracked move history for all mons.

![image](https://user-images.githubusercontent.com/4258818/208654063-68095bd5-ad11-42ac-bf34-01d73c68e72d.png)

**Comment by UTDZac** [comment]
Feature is done and available in version 7.1.0.

**Comment by UTDZac** [comment]
Here's another scenario where seeing multiple different levels of the same Pokémon causes the "4 move window" display on the main Tracker screen to fail to understand how to appropriately show all possible moves.

In this case, i believe (mostly guessing) the last few Altaria encounters were:

1. 32
2. 27
3. 33
4. 30
5. 33

![image](https://user-images.githubusercontent.com/4258818/209289712-6c61a493-f3b3-4aa7-bd06-5e1d9b2830fb.png)
